mesonbuild/dependencies/dub.py

```diff
@@ -432,3 +528,32 @@ def find() -> T.Optional[T.Tuple[ExternalProgram, str]]:
                      '(version %s)' % dubver)
 
         return found
+
+    def _find_dub_home(self) -> str:
+        # first check environment
+        dubhome = os.environ.get('DUB_HOME')
+        if dubhome is not None:
+            return dubhome
+        dpath = os.environ.get('DPATH')
+        if dpath is not None:
+            return os.path.join(dpath, 'dub')
+
+        # default value
+        is_windows = self.env.machines.host.is_windows()
+        if is_windows:
+            appdata = os.environ.get('LOCALAPPDATA')
+            if appdata is None:
+                # LOCALAPPDATA is not defined before Vista
+                appdata = os.environ.get('APPDATA')
+            assert (isinstance(appdata, str))
+            return os.path.join(appdata, 'dub')
+        else:
+            home = os.environ.get('HOME')
+            assert (isinstance(home, str))
+            return os.path.join(home, '.dub')
+
+    def _find_cache_dir(self) -> str:
+        dubhome = self._find_dub_home()
+        # TODO: handle (rare) case where dubhome is overriden
+        # in Dub's settings file.
+        return os.path.join(dubhome, 'cache')
```

Comment:
This is to make easier a future change to take into account the user customization of dubhome location in the settings (JSON files that Dub reads in various places)

Reply:
Does that just change what the dubhome is while leaving the contents such as cache/ alone?

Reply:
The user can customize the location of dubhome in its personal settings.
The cache root is always `[dubhome]/cache`.

Reply:
It is not mandatory for Meson to handle this. In case of dependency not found, I can add to the error message the cache location we're searching in to indicate to the user that Meson will not try other cache location